src/main/java/com/google/cloud/tools/gradle/appengine/appyaml/AppEngineAppYamlPlugin.java

```diff
@@ -89,10 +91,10 @@ private void configureExtensions() {
           // we can only set the default location of "archive" after project evaluation (callback)
           if (stageExtension.getArtifact() == null) {
             if (project.getPlugins().hasPlugin(WarPlugin.class)) {
-              War war = (War) project.getProperties().get("war");
+              War war = tasks.withType(War.class).getByName("war");
               stageExtension.setArtifact(war.getArchivePath());
             } else if (project.getPlugins().hasPlugin(JavaPlugin.class)) {
-              Jar jar = (Jar) project.getProperties().get("jar");
+              Jar jar = tasks.withType(Jar.class).getByName("jar");
```

Comment:
Same here: it is better to use `tasks.named(...)`

Reply:
But the code below needs to do `Jar.getArchivePath()`, which I believe will require instantiating the task anyway, right?

Reply:
Oh sorry, nvm. Didn't see the other comment.